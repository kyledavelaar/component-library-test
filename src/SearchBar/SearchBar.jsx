// @flow

import * as React from 'react';
import s from './SearchBar.scss';
import SearchIcon from './search.svg';

type Props = {
  placeholder: string,
  query: string,
  onSearch: (val: string) => void,
  onChange: (val: string) => void,
}

type State = {
}

export class SearchBar extends React.PureComponent<Props, State> {
  state = {

  }

  render() {
    const { placeholder, query } = this.props;

    return (
      <div id={s.wrapper}>
        <img
          src={SearchIcon}
          alt="search"
          className={s.icon}
          onClick={() => this.onSearchClick()}
        />
        <input
          className={s.search_input}
          type="text"
          value={query}
          onKeyUp={(e) => this.onInputChange(e)}
          onChange={(e) => this.onInputChange(e)}
          placeholder={placeholder}
        />
      </div>
    )

  }


///////////////////////////////////////////////////////////
//  EVENT HANDLERS
//////////////////////////////////////////////////////////

  onInputChange(e: any) {
    e.preventDefault();
    const query = e.currentTarget.value;
    this.props.onChange(query);

    if (e.keyCode === 13) {
      this.props.onSearch(query);
    }
  }

  onSearchClick() {
    this.props.onSearch(this.props.query);
  }

}



export default SearchBar
