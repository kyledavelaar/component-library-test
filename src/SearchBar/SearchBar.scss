@import '../variables.scss';

#wrapper {
  margin: 10px;
  position: relative;
}

.icon {
  color: $graphite;
  position: absolute;
  top: 10px;
  cursor: pointer;
}

.search_input {
  width: 300px;
  overflow: auto;
  border: none;
  border-bottom: 1px solid $dustygrey;
  font-size: 14px;
  padding: 0px 20px;
  background: transparent;
}
