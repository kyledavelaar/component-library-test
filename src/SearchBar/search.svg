<?xml version="1.0" encoding="UTF-8"?>
<svg width="13px" height="13px" viewBox="0 0 13 13" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 52.3 (67297) - http://www.bohemiancoding.com/sketch -->
    <title>13078B34-836E-443E-917D-555F83F396EA</title>
    <desc>Created with sketchtool.</desc>
    <defs>
        <path d="M106.832952,1174.02094 C107.055247,1174.24319 107.055247,1174.60826 106.835554,1174.82786 C106.727785,1174.93976 106.579991,1175 106.428087,1175 C106.278912,1175 106.13151,1174.93878 106.023222,1174.83052 L103.086942,1171.89479 C102.101387,1172.69699 100.870976,1173.14402 99.573067,1173.14402 C96.5008603,1173.14402 94,1170.64363 94,1167.57201 C94,1164.49944 96.4990136,1162 99.573067,1162 C102.646223,1162 105.146134,1164.49854 105.146134,1167.57201 C105.146134,1168.86967 104.699023,1170.09985 103.896673,1171.08522 L106.832952,1174.02094 Z M95.1459688,1167.57201 C95.1459688,1170.01201 97.1317684,1171.99827 99.5702714,1171.99827 C102.009604,1171.99827 103.994574,1170.01119 103.994574,1167.57201 C103.994574,1165.1334 102.010172,1163.14855 99.5702714,1163.14855 C97.1311993,1163.14855 95.1459688,1165.13257 95.1459688,1167.57201 Z" id="path-1"></path>
        <filter x="-15.4%" y="-15.4%" width="130.8%" height="130.8%" filterUnits="objectBoundingBox" id="filter-2">
            <feGaussianBlur stdDeviation="1.5" in="SourceAlpha" result="shadowBlurInner1"></feGaussianBlur>
            <feOffset dx="0" dy="1" in="shadowBlurInner1" result="shadowOffsetInner1"></feOffset>
            <feComposite in="shadowOffsetInner1" in2="SourceAlpha" operator="arithmetic" k2="-1" k3="1" result="shadowInnerInner1"></feComposite>
            <feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.5 0" type="matrix" in="shadowInnerInner1"></feColorMatrix>
        </filter>
    </defs>
    <g id="CP-Style-Guide" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g transform="translate(-94.000000, -1162.000000)" fill-rule="nonzero" id="search">
            <g>
                <use fill="#333333" xlink:href="#path-1"></use>
                <use fill="black" fill-opacity="1" filter="url(#filter-2)" xlink:href="#path-1"></use>
            </g>
        </g>
    </g>
</svg>