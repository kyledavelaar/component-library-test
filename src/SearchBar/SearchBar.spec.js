import React from 'react';
import { mount, shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import { SearchBar } from './SearchBar';

Enzyme.configure({ adapter: new Adapter() })


describe('SearchBar renders',() => {

  it('SearchBar component created', () => {
    const rendered = renderer.create(
      <SearchBar
        placeholder="search..."
        query="hi there"
      />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  })
})