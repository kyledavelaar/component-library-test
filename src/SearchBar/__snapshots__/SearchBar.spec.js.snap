// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders SearchBar component created 1`] = `
<div
  id={undefined}
>
  <img
    alt="search"
    className={undefined}
    onClick={[Function]}
    src="search.svg"
  />
  <input
    className={undefined}
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="search..."
    type="text"
    value="hi there"
  />
</div>
`;
